Fix quorum pre-image unit test

hewison-chris opened the pull request:
This test has often failed recently. See commits for details on fix.

hewison-chris committed:
QuorumPreimage unit test: Do not hardcode the max listeners

The max listeners should be the sum of the validators, fullnodes and
registry. In `base.d` it is correctly calculated so just remove the
`testConf` hardcoded setting.

hewison-chris committed:
QuorumPreimage unit test: Assert all enrollemnts are included

The test relies on the fact that all Genesis and outsider validators
are enrolled in the last block of the cycle. Let's just assert this is
the case. Maybe we will need to refactor the test if this assert fails.

hewison-chris committed:
QuorumPreimage unit test: Remove sleep and wait

These steps should not be required as the next step has retries.

hewison-chris committed:
Trivial comment correction

hewison-chris committed:
QuorumPreimage unit test: Do not use GenerateBlocks

For blocks 21 -> 39 it is better to send the tx to one client and check
for the height in the test code as there are outsider validators. In the
`generateBlocks` function it checks that the txs are gossiped to all the
validators before setting the time for the next block. But as the
outsiders were not enrolled until block 20 the client may gossip
the tx before it has updated the peers it sends to. Normally this does
not matter because once an envelope is recieved by the newly enrolled
node it will try to fetch it.

linked0 reviewed (approved):
LGTM

hewison-chris commented:
This test is never failing locally so is really hard to fix at the moment.

linked0 commented:
> This test is never failing locally so is really hard to fix at the moment.

If this fails again on CI even with this change, How about solving this problem with other approaches later.

linked0 commented:
But if you think there are any improvements in the test result, it's not bad to merge it.

hewison-chris committed:
QuorumPreimage unit test: use auto enroll for validators

If we use auto enroll then we do not need to manually enroll the
outsiders for their second enrollment.

hewison-chris committed:
Trivial debug logging updates

hewison-chris committed:
QuorumPreimage test: Reduce quorum threshold
